gitconfig color.ui = always breaks patches

With git's `color.ui` set to `always`, I get the following errors:
```
[ERROR] Could not update repository "bb-overlay-fork": "/usr/local/bin/patch -p1 -i /Users/pgiarrusso/.opam/log/processed-patch-58446-70a526" exited with code 2 "/usr/local/bin/patch: **** Only garbage was found in the patch input."
[bb-overlay] synchronised from git+https://github.com/Blaisorblade/opam-overlay.git
[ERROR] Could not update repository "bb-overlay": "/usr/local/bin/patch -p1 -i /Users/pgiarrusso/.opam/log/processed-patch-58446-e80101" exited with code 2 "/usr/local/bin/patch: **** Only garbage was found in the patch input."
[iris-dev] synchronised from git+https://gitlab.mpi-sws.org/iris/opam.git
[ERROR] Could not update repository "iris-dev": "/usr/local/bin/patch -p1 -i /Users/pgiarrusso/.opam/log/processed-patch-58446-8b8e2c" exited with code 2 "/usr/local/bin/patch: **** Only garbage was found in the patch input."
```

Maybe `opam` needs to override `color.ui` git settings, like it does (IIRC) for a few others (e.g. https://github.com/ocaml/opam/pull/3788)?

Tested on opam 2.0.8.

--- a/master_changes.md
+++ b/master_changes.md
@@ -91,6 +91,7 @@ users)
 
 ## VCS
   * Use 4.08's unnamed functor arguments to silence warning 67 [#4775 @dra27]
+  * git: disable colored output [#4884 @rjbou]
 
 ## Build
   * Bump src_exts and fix build compat with Dune 2.9.0 [#4752 @dra27]

--- a/src/repository/opamGit.ml
+++ b/src/repository/opamGit.ml
@@ -50,6 +50,7 @@ module VCS : OpamVCS.VCS = struct
        *)
       git repo_root [ "config" ; "--local" ; "core.autocrlf"; "false"];
       git repo_root [ "config" ; "--local" ; "core.eol"; "lf"];
+      git repo_root [ "config" ; "--local" ; "color.ui"; "false" ];
       (* Document the remote for user-friendliness (we don't use it) *)
       git repo_root [ "remote"; "add"; "origin"; OpamUrl.base_url repo_url ];
     ] @@+ function

